node/internal/node_utils.bzl

```diff
@@ -118,31 +135,41 @@ def node_install(ctx, install_path, modules):
+            wName = demangle_package_name(w.name)
             cmds += [
-                "mkdir -p %s/%s/node_modules/%s" % (install_path.path, m.name, w.name),
-                "tar -xzf %s -C %s/%s/node_modules/%s --strip-components 1" % (w.file.path, install_path.path, m.name, w.name),
+                "mkdir -p %s/%s/node_modules/%s" % (install_path.path, mName, wName),
+                "tar -xzf %s -C %s/%s/node_modules/%s --strip-components 1" % (w.file.path, install_path.path, mName, wName),
             ]
             for w2 in w.wrapped_deps:
                 inputs += [w2.file]
+                w2Name = demangle_package_name(w2.name)
                 cmds += [
-                    "mkdir -p %s/%s/node_modules/%s/node_modules/%s" % (install_path.path, m.name, w.name, w2.name),
-                    "tar -xzf %s -C %s/%s/node_modules/%s/node_modules/%s --strip-components 1" % (w2.file.path, install_path.path, m.name, w.name, w2.name),
+                    "mkdir -p %s/%s/node_modules/%s/node_modules/%s" % (install_path.path, mName, wName, w2Name),
+                    "tar -xzf %s -C %s/%s/node_modules/%s/node_modules/%s --strip-components 1" % (w2.file.path, install_path.path, mName, wName, w2Name),
                 ]
                 for w3 in w2.wrapped_deps:
                     inputs += [w3.file]
+                    w3Name = demangle_package_name(w3.name)
                     cmds += [
-                        "mkdir -p %s/%s/node_modules/%s/node_modules/%s/node_modules/%s" % (install_path.path, m.name, w.name, w2.name, w3.name),
-                        "tar -xzf %s -C %s/%s/node_modules/%s/node_modules/%s/node_modules/%s --strip-components 1" % (w3.file.path, install_path.path, m.name, w.name, w2.name, w3.name),
+                        "mkdir -p %s/%s/node_modules/%s/node_modules/%s/node_modules/%s" % (install_path.path, mName, wName, w2Name, w3Name),
+                        "tar -xzf %s -C %s/%s/node_modules/%s/node_modules/%s/node_modules/%s --strip-components 1" % (w3.file.path, install_path.path, mName, wName, w2Name, w3Name),
                     ]
-                    if w2.wrapped_deps: fail("nested wrapped dependencies not supported deeper than 3 levels (in %s)" % w3.label)
+                    for w4 in w3.wrapped_deps:
+                        inputs += [w4.file]
+                        w4Name = demangle_package_name(w4.name)
+                        cmds += [
+                            "mkdir -p %s/%s/node_modules/%s/node_modules/%s/node_modules/%s/node_modules/%s" % (install_path.path, mName, wName, w2Name, w3Name, w4Name),
+                            "tar -xzf %s -C %s/%s/node_modules/%s/node_modules/%s/node_modules/%s/node_modules/%s --strip-components 1" % (w4.file.path, install_path.path, mName, wName, w2Name, w3Name, w4Name),
+                        ]
+                    if w3.wrapped_deps: fail("nested wrapped dependencies not supported deeper than 4 levels (in %s)" % w4.label)
```

Comment:
Suspect this should be w4.wrapped_deps - existing bug had w2... Maybe going to 4 levels wasn't required?

Reply:
I think you're right, I just needed to change it to `w3.wrapped_deps` in the original code. Should we keep support for 4 levels?